src/codemod.py

```diff
@@ -81,20 +81,45 @@
 
 import sys, os
 
+def matches_extension(path, extension):
+  """
+  Returns True if path has the given extension, or if
+  the last path component matches the extension.
+
+  >>> matches_extension("./www/profile.php", "php")
+  True
+  >>> matches_extension("./scripts/menu.js", "html")
+  False
+  >>> matches_extension("./LICENSE", "LICENSE")
+  True
+  """
+  root, ext = os.path.splitext(path)
+  if ext == '':
+    # If there is not extension, grab the file name and
+    # compare it to the given extension.
+    file_name = os.path.split(root)[1]
+    return file_name == extension
+  else:
+    # If the is an extension, drop the leading period and
+    # compare it to the extension.
+    return ext[1:] == extension
+
 def path_filter(extensions=None, exclude_paths=[]):
   """
   Returns a function (useful as the path_filter field of a Query instance)
-  that returns True iff the path it is given has an extension one of the
+  that returns True if the path it is given has an extension one of the
```

Comment:
I'll revert this. I found the "iff" a bit misleading, since it doesn't state the full set of conditions necessary to match--not only does the path have to match one of the extensions provided, it also must not be in an excluded directory.

But it's not wrong: it'll match iff its extension matches (and some other conditions).